### PR TITLE
Add ask_password and tests

### LINKED ISSUE #16: Feature request: ask_password
This package is beautiful, and I would love to switch all my input to it.

I didn't see a way to ask for a password though. It would be the same as `ask_string`, but with either no echo, or a mask.

My apologies if the feature already exists, I'll look into creating a PR in a couple days if this is something you are ok with.

### PR DESCRIPTION
**Pull request by drazisil**
* Add an ask_password method
* Add a read_password method
* Add tests for both
* Closes #16

### CODE REVIEW

File: ui/tests/test_ui.py [hunk 1]
```diff
@@ -107,6 +112,12 @@ def test_ask_string():
         assert res == "milk"
 
 
+def test_ask_password():
+    with mock.patch('getpass.getpass') as m:
+        m.side_effect = ["chocolate!", ""]
+        res = ui.ask_password("gulty pleasure?")
```
Comment (dmerejkowsky) [inline ui/tests/test_ui.py]: I think you mean 'guilty' :)

**Review by dmerejkowsky — APPROVED** [review]

**Comment by dmerejkowsky** [comment]
Nice PR:)

Could you please patch the documentation too ? You can add `ask_password` at the end of the 'Ask for user input' section.

File: ui/tests/test_ui.py (hunk 1 above)
Comment (drazisil) [reply ui/tests/test_ui.py]: Spelling, bah! 😉

**Comment by drazisil** [comment]
Wasn't sure if the docs were automagically generated. They have been updated, thanks!

**Comment by dmerejkowsky** [comment]
Thanks :)

**Pull request merged**